src/services/weatherService.js

```diff
@@ -58,6 +61,68 @@ class WeatherService {
       console.error(e.message);
     }
   }
+
+  async getHourlyForecast(latitude, longitude, selectedDate) {
+    try {
+      const params = {
+        latitude,
+        longitude,
+        hourly: `
+          temperature_2m,
+          relative_humidity_2m,
+          dew_point_2m,
+          apparent_temperature,
+          precipitation,
+          weather_code,
+          surface_pressure,
+          cloud_cover,
+          visibility,
+          evapotranspiration,
+          vapour_pressure_deficit,
+          wind_speed_10m,
+          wind_direction_10m,
+          wind_gusts_10m
+        `
+          .trim()
+          .replace(/\s+/g, ''),
+        timezone: 'auto',
+      };
+      const now = DateTime.now();
+
+      const { data } = await axios.get(`${baseUrl}/forecast`, { params });
+
+      return data.hourly.time
+        .filter(time => time.startsWith(selectedDate))
+        .map((time, index) => ({
+          time,
+          temperature: data.hourly.temperature_2m[index],
+          feelsLike: data.hourly.apparent_temperature[index],
+          dewPoint: data.hourly.dew_point_2m[index],
+          precipitation: data.hourly.precipitation[index],
+          windSpeed: data.hourly.wind_speed_10m[index],
+          windDirection: data.hourly.wind_direction_10m[index],
+          windGusts: data.hourly.wind_gusts_10m[index],
+          pressure: data.hourly.surface_pressure[index],
+          humidity: data.hourly.relative_humidity_2m[index],
+          cloudCover: data.hourly.cloud_cover[index],
+          visibility: data.hourly.visibility[index],
+          evapotranspiration: data.hourly.evapotranspiration[index],
+          vpd: data.hourly.vapour_pressure_deficit[index],
+          weatherCode: data.hourly.weather_code[index],
+        }))
+        .filter(data => DateTime.fromISO(data.time) > now)
```

Comment:
can we move this filter to the first one? this might reduce complexity a bit an we could also remove second map in that case